src/qn_model.jl

```diff
@@ -3,13 +3,13 @@ export QuasiNewtonModel, LBFGSModel, LSR1Model
 abstract type QuasiNewtonModel <: AbstractNLPModel end
 
 mutable struct LBFGSModel <: QuasiNewtonModel
-  meta :: NLPModelMeta
+  meta :: AbstractNLPModelMeta
```

Comment:
Isn't it discouraged to use abstract types inside structs?

Reply:
I can make it parametric. But notice that there is an `AbstractNLPModel` below.

Reply:
Hmm, that's an idea. I wonder what it would break.

Reply:
We're about to find out.